### PR TITLE
Add Travis CI integration with repo.

### PR DESCRIPTION
**Pull request by philliphoff**
Adds Travis CI integration with this repository.  On any push or PR, Travis CI will clone, install, and build the repository.  Travis CI will use the same Gulp tasks (`install` and `default`) of the root pseudo-package.

Note: The use of TSD for obtaining TypeScript definition files incurs additional requests during the build that may hit GitHub rate limits (for anonymous accounts).  Hence, I am using a personal GitHub account token to allow an increased rate limit (assigned to the environment variable TSD_GITHUB_TOKEN configured in Travis CI).

### CODE REVIEW

**Comment by philliphoff** [comment]
Associated with #40.

**Comment by philliphoff** [comment]
Minor note: the badge shown in the README will always reflect the build state of the `master` branch, regardless of the branch one views.  It doesn't appear that GitHub provides a static means for Travis CI to identify which branch's README is being viewed in order to serve the correct status.  This seems to be an oft requested integration feature between GitHub and Travis CI.

The only way (today) to have branch-specific statuses is to permanently "fork" the README in each branch.  This is probably too big of a merge hassle for our typical feature branches but may be necessary if/when we move to parallel master/development branches.  (In that case still, anytime a merge is done, one will need to be careful to not modify the status link.)

**Comment by philliphoff** [comment]
I've also added integration with Slack, so Travis CI builds will get reported in the #notifications channel.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+install:
+- npm install -g gulp
+- npm install -g tsd
+- npm install
+- gulp install
+script:
+- gulp
+notifications:
+  slack:
+    secure: QFZ33N9mqS+VOqwTyinC/avLk1UMmw0QF6013iJUSB+hDJipncvAPxaHdRzGrKyapEYsl6wc68f0euBuLYA1l7HiaJBLrnrgjKuaMqHcPJhYbmbIeDxxlIMaMv2rY5j4yhiwaWx8UAoaKWBTmrn85mlQC8UI9ugaPfS5T2+CUsOD99y6DoEW1wIYQftJb/VA4XYQKVGGFR9rjwaSTTWmjmfjn59jh9UMYh29toHMlkJPqFAoD8KmFh3qUiQw+Wppqaf2q+EDXp+31AZdorm1wRB3ioPi3t/iv0Mic4OQvDT9Qp/1xkPixVeNywHHcxWzdwAe5Ey3+TEgGWd/vo/nwQjBq1Ug2e5o4ApLxIGL1thARnx7NdOS+pRHnmJ6xfbppIurGU26XKqNGe1pOGJ1VYyH0886N+wM5CUxCjyZsjOLQcWZ0B95nxBgBLZ7RNcG0tVUeYLzZMIZY30Bgmi7k7EC5EnSghTUrP+Kiq8JoAEnw/JmYIthnnDEBpYsiklC9qRaiKZv+1mT6a7Ga5qM5SlIqLpQTHh7XZorCjj92KxvQghscw+K9z/o9goopJULVL34ybJERneTOy5i6eO4FondiwSDcjhlKUEZ8o7bDHppAsFomlnr1eQvp4e5jGK9465/2hKRlQ/Ciu19xx3E1MVDWoBlryEcWAhf3zWgQAg=
```
Comment (mike-kaufman) [inline .travis.yml]: Is this OK to put into the public domain?  what rights does some have with this block?

Comment (philliphoff) [inline .travis.yml]: Yes, the `secure` element is repo-specific encryption used by Travis CI for the purposes of storing sensitive information within the Travis configuration file, and is the recommended way of doing so.

Comment (philliphoff) [inline .travis.yml]: Here's a [link](https://docs.travis-ci.com/user/encryption-keys/) to the Travis CI documentation.

Comment (avanderhoorn) [inline .travis.yml]: Just our of curiosity what type of information we storing in it.

**Comment by avanderhoorn** [comment]
:shipit:

File: .travis.yml (hunk 1 above)
Comment (philliphoff) [inline .travis.yml]: It's just the encrypted Slack team/channel and auth token that allows Travis CI to post to our Glimpse notifications Slack channel.

**Pull request merged**